test/validation/api/shmem/shmem.c

```diff
@@ -212,6 +214,55 @@ void shmem_test_basic(void)
 	CU_ASSERT(0 == odp_shm_free(shm));
 }
 
+/*
+ * maximum size reservation
+ */
+static void shmem_test_max_reserve(void)
+{
+	odp_shm_capability_t capa;
+	odp_shm_t shm;
+	uint64_t size, align;
+	uint8_t *data;
+	uint64_t i;
+
+	memset(&capa, 0, sizeof(odp_shm_capability_t));
+	CU_ASSERT_FATAL(odp_shm_capability(&capa) == 0);
+
+	CU_ASSERT(capa.max_blocks > 0);
+
+	size  = capa.max_size;
+	align = capa.max_align;
+
+	/* Assuming that system has at least MAX_MEMORY_USED bytes available */
+	if (capa.max_size == 0 || capa.max_size > MAX_MEMORY_USED)
+		size = MAX_MEMORY_USED;
+
+	if (capa.max_align == 0 || capa.max_align > MEGA)
+		align = MEGA;
```

Comment:
Shouldn't `MEGA` be a power of 2 for alignment purposes? I.e., 1024 x 1024 rather than 1000 x 1000? And if the implementation supports an even higher `max_align` why not test that as well?

Reply:
Very large align could result very large allocation and thus again system run out of memory (e.g. 1TB align => >1TB alloc).

OK. I'll change align max to be a power of two.

Reply:
Same story as for `capa.max_size`. I'd expect most implementations to return `capa.max_align` to be either 0 or some reasonable value like 4K or 1M. However, if they specify something else then they should be able to deliver that.

Reply:
Maybe with 1GB hugepages the max align is 1GB, and 16GB hugepages 16GB, ...

Reply:
Again, what does the application gain from having a non-zero `max_align` returned if it's unable to make use of it?

Reply:
If max size is 4GB, max align is 1GB. A resevation may consume e.g. 4.5GB memory. Again it depends on other applications and ODP instances if such amount of system memory is still free when application tries to reserve it. Maybe first couple instances succeeded, but then system did run out of memory.